File: js/isNarcissistic.js
const isNarcissistic = n => {
  // turn n into string to get length
  // and split into indvidual nums array
  // for multiplication
  const numberLength = n.toString().length;
  let numbers = n.toString().split("");
  let sum = [];

  // go through numbers array, multiply by
  // length exponent, push to sum array
  for (let i = 0; i < numberLength; i++) {
    sum.push(Math.pow(parseInt(numbers[i], 10), numberLength));
  }

  // reduce sum array to get total sum
  sum = sum.reduce((a, b) => a + b);

  // if equal to og, return true
  // else not armstrong
  if (sum === n) {
    return true;
  } else {
    return false;
  }
};

File: python/is_narcissistic.py
def is_narcissistic(i):
    # turn int into string
    number_string = str(i)
    sum_of_digits = 0
    # iterate over string
    for num in number_string:
        # multiply each num by length exponent
        sum_of_digits += int(num) ** len(number_string)
    # if sum of digits to power of n equals i, it's narcissistic
    if sum_of_digits == i:
        return True
    else:
        return False

